test/helper.js

```diff
@@ -1,82 +1,184 @@
-let assert      = require('assert');
-let path        = require('path');
-let _           = require('lodash');
-let api         = require('../src/api');
-let taskcluster = require('taskcluster-client');
-let mocha       = require('mocha');
-let exchanges   = require('../src/exchanges');
-let load        = require('../src/main');
-let config      = require('typed-env-config');
-let testing     = require('taskcluster-lib-testing');
-
-// Load configuration
-let cfg = config({profile: 'test'});
-
-let testclients = {
+const assert = require('assert');
+const path = require('path');
+const _ = require('lodash');
+const builder = require('../src/api');
+const data = require('../src/data');
+const taskcluster = require('taskcluster-client');
+const mocha = require('mocha');
+const exchanges = require('../src/exchanges');
+const load = require('../src/main');
+const config = require('typed-env-config');
+const {stickyLoader, Secrets, fakeauth} = require('taskcluster-lib-testing');
+
+const testclients = {
   'test-client': ['*'],
   'test-server': ['*'],
 };
 
-// Create and export helper object
-let helper = module.exports = {};
-
-// Skip tests if no credentials is configured
-if (!cfg.pulse || !cfg.taskcluster) {
-  console.log('Skip tests due to missing credentials!');
-  process.exit(1);
-}
-
-// Configure PulseTestReceiver
-helper.events = new testing.PulseTestReceiver(cfg.pulse, mocha);
-
-// Hold reference to authServer
-let authServer = null;
-let webServer = null;
-
-// Setup before tests
-mocha.before(async () => {
-  // Create mock authentication server
-  testing.fakeauth.start(testclients);
-
-  await load('expire-cache-purges', {profile: 'test', process: 'test'});
-  webServer = await load('server', {profile: 'test', process: 'test'});
-
-  // Create client for working with API
-  helper.baseUrl = 'http://localhost:' + webServer.address().port + '/v1';
-  let reference = api.reference({baseUrl: helper.baseUrl});
-  helper.PurgeCache = taskcluster.createClient(reference);
-  // Utility to create an PurgeCache instance with limited scopes
-  helper.scopes = (...scopes) => {
-    helper.purgeCache = new helper.PurgeCache({
-      // Ensure that we use global agent, to avoid problems with keepAlive
-      // preventing tests from exiting
-      agent:            require('http').globalAgent,
-      baseUrl:          helper.baseUrl,
+exports.suiteName = path.basename;
+exports.rootUrl = 'http://localhost:60415';
+
+exports.load = stickyLoader(load);
+
+suiteSetup(async function() {
+  exports.load.inject('profile', 'test');
+  exports.load.inject('process', 'test');
+});
+
+// set up the testing secrets
+exports.secrets = new Secrets({
+  secretName: 'project/taskcluster/testing/taskcluster-purge-cache',
+  secrets: {
+    azure: [
+      {env: 'AZURE_ACCOUNT', cfg: 'azure.accountId', name: 'accountId'},
+      {env: 'AZURE_ACCOUNT_KEY', cfg: 'azure.accessKey', name: 'accountKey'},
```

Comment:
This doesn't appear to be used..  In tests, I just hardcoded the accountId [in config.yml](https://github.com/taskcluster/taskcluster-hooks/blob/master/config.yml#L61) and lef the Taskcluster credentials do the rest.